scripts/common-queue.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -m
+
+DOCKER_ID=
+SLEEP_PID=
+
+stop_sleep() { [ -z "$SLEEP_PID" ] && return; kill -TERM $SLEEP_PID; exit 0; }
+stop_container() { [ -z "${DOCKER_ID}" ] && return; docker kill --signal=SIGTERM ${DOCKER_ID}; }
+
+start() {
+  TEST_TYPE=$1
+  NEXT_REVISION_CMD=$2
+  shift 2
+
+  running=1
+  trap '' HUP
+  trap 'running=0; stop_sleep; stop_container' INT TERM
+
+  while [ $running -eq 1 ]
+  do
+    while true
+    do
+      PREV_SDK_REVISION=${SDK_REVISION}
+      SDK_REVISION=$(${NEXT_REVISION_CMD})
+      [ -z "$SDK_REVISION" ] && exit 1
+      OUTPUT_DIR="${TEST_TYPE}-${SDK_REVISION}"
+      [ ! -d "${OUTPUT_DIR}" ] && break
+      [ "x$SDK_REVISION" != "x$PREV_SDK_REVISION" ] && continue
+      sleep 60 &
+      SLEEP_PID=$!
+      wait $SLEEP_PID
+      SLEEP_PID=
+    done
+    echo "processing ${SDK_REVISION}"
+    mkdir "${OUTPUT_DIR}"
+    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod -v loadgen-yarn-cache:/home/node/.cache/yarn -v "$(pwd)/${OUTPUT_DIR}:/out" -e SDK_REVISION=${SDK_REVISION} -e SDK_REPO=${SDK_REPO} --name "${OUTPUT_DIR}" loadgen-runner --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```

Comment:
```suggestion
    DOCKER_ID=$(docker create -v loadgen-go-pkg-mod:/go/pkg/mod \
      -v loadgen-yarn-cache:/home/node/.cache/yarn \
      -v "$(pwd)/${OUTPUT_DIR}:/out" \
      -e SDK_REVISION=${SDK_REVISION} \
      -e SDK_REPO=${SDK_REPO} \
      --name "${OUTPUT_DIR}" \
      loadgen-runner \
      --test-data.test-type=${TEST_TYPE} "$@") || exit $?
```

Reply:
Resolved in 6287b8096ebd48bb08671feb6c24c422e917dd3f